chain/ethereum/src/block_stream.rs

```diff
@@ -398,6 +398,17 @@ where
                             // Calculate the range size according to the target number of triggers,
                             // respecting the global maximum and also not increasing too
                             // drastically from the previous block range size.
+                            //
+                            // An example of the block range dynamics:
+                            // - Start with a block range of 1, target of 1000.
+                            // - Scan 1 block:
+                            //   0 triggers found, max_range_size = 10, range_size = 10
+                            // - Scan 10 blocks:
+                            //   2 triggers found, 0.2 per block, range_size = 1000 / 0.2 = 5000
+                            // - Scan 5000 blocks:
+                            //   500 triggers found, 0.1 per block, range_size = 500 / 0.2 = 2500
```

Comment:
Hmm. If I scan 5000 blocks and I find 500 triggers but my target is 1000, shouldn't I increase the block range to 10000, not keep it at 500?

Similar with scanning 2500 blocks and finding 500 triggers. In that case I'd also think that I have to double my range size to get to the target amount of triggers in the range.

Reply:
Right this is totally wrong, I think I fixed it.